subprojects/luajit/src/host/meson.build

```diff
@@ -11,25 +11,33 @@ endif
 dynasm_dasc = files('../vm_@0@.dasc'.format(dynasm_arch))
 dasm = [minilua, files('../../dynasm/dynasm.lua')]
 
-# BUG: meson does not resolve paths correctly for subprojects
-hpre = '#include "@0@/../lj_arch.h"'.format(meson.current_source_dir())
-
-checkdefs = [
-    ['LJ_ARCH_BITS', '64', ['-D', 'P64']],
-    ['LJ_HASJIT', '1', ['-D', 'JIT']],
-    ['LJ_HASFFI', '1', ['-D', 'FFI']],
-    ['LJ_DUALNUM', '1', ['-D', 'DUALNUM']],
-]
-
-if host_machine.cpu_family() == 'x86'
-    checkdefs += ['__SSE2__', '1', ['-D', 'SSE']]
-endif
-
-foreach def: checkdefs
-    if cc.get_define(def[0], prefix: hpre) == def[1]
-        dasm += def[2]
+if cc.get_id() == 'msvc'
+    # the cl.exe preprocessor seemingly removes/expands macros, so hardcode flags
+    dasm += ['-D', 'WIN', '-D', 'JIT', '-D', 'FFI']
```

Comment:
No dualnum? It seems weird to have this be different between msvc and everywhere else. I wonder if we just shouldn't be setting it at all? What did Aegisub do previously?

Reply:
If you're using the Makefile to compile on x86(_64), it seems that DUALNUM only gets set if you explicitly set LUAJIT_NUMMODE to 2, meaning the corresponding get_define will return false by default (which I can confirm, DUALNUM is not present in my build.ninja on Linux). When compiling for MSVC using msvcbuild.bat you simply don't get the option at all. I don't see any mention of the option in the Aegisub VS projects.

Reply:
Sounds like we can just get rid of it entirely. If [this](https://news.ycombinator.com/item?id=9989605) is accurate, I don't think we have any need for it.

Reply:
I've removed it now.